shared/search/result-row/container.js

```diff
@@ -1,42 +1,54 @@
 // @flow
-import SearchResultRow from '.'
-import {Map} from 'immutable'
+import SearchResultRow, {type Props} from '.'
 import {userIsActiveInTeamHelper} from '../../constants/teams'
-import {followStateHelper} from '../../constants/search'
+import {followStateHelper, makeSearchResult} from '../../constants/search'
 import {type SearchResultId} from '../../constants/types/search'
 import {connect, type TypedState, setDisplayName, compose} from '../../util/container'
 
-type OwnProps = {
+export type OwnProps = {|
   disableIfInTeamName: ?string,
   id: SearchResultId,
+  selected: boolean,
   onClick: () => void,
   onMouseOver?: () => void,
   onShowTracker?: () => void,
-}
+|}
 
-const mapStateToProps = (
-  state: TypedState,
-  {disableIfInTeamName, id, onClick, onMouseOver, onShowTracker}: OwnProps
-) => {
-  const result: any = state.entities.getIn(['search', 'searchResults', id], Map()).toObject()
+const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
+  const result = state.entities.search.searchResults.get(ownProps.id, makeSearchResult())
```

Comment:
should make this a constant variable else it'll make a new record each time and render thrash
aka
const emptySearch = makeSearchResult() // in the module
searchResults.get(ownProps.id, emptySearch)

Reply:
d'oh, done.